nlu/models/intent_classifier.py

```diff
@@ -119,35 +122,70 @@ def format_scores(self, scores: Dict[str, dict]):
     @tfbp.runnable
     def fit(self):
         """Training"""
-        encoded_texts, encoded_intents, encoded_slots, intent_names, slot_names = self.data_loader(
-            self.tokenizer)
-
-        if self.hparams.intent_num_labels != len(intent_names):
-            raise ValueError(
-                f"Hyperparam intent_num_labels mismatch, should be : {len(intent_names)}"
-            )
-
-        # Hyperparams, Optimizer and Loss function
-        opt = Adam(learning_rate=3e-5, epsilon=1e-08)
-
-        losses = SparseCategoricalFocalLoss(gamma=self.hparams.gamma)
-
-        metrics = [SparseCategoricalAccuracy("accuracy")]
-
-        # Compile model
-        self.compile(optimizer=opt, loss=losses, metrics=metrics)
-
-        x = {"input_ids": encoded_texts["input_ids"], "token_type_ids": encoded_texts["token_type_ids"],
-             "attention_mask": encoded_texts["attention_mask"]}
-
-        super().fit(
-            x, encoded_intents, epochs=self.hparams.num_epochs, batch_size=32, shuffle=True)
-
-        # Persist the model
-        self.extra_params["intent_names"] = intent_names
-
-        self.save()
-
+        # Start MLflow run
+        with mlflow.start_run() as run:
```

Comment:
Shouldn't this be implement in the parent class so that it would work for all models ?

Reply:
Ideally it should be implemented as a helper class. Otherwise, we're logging metrics during training. The base class doesn't have a fit method.